pkg/controller/nodes/task/resourcemanager/redis_resourcemanager.go

```diff
@@ -184,16 +175,52 @@ func (r *RedisResourceManager) GetID() string {
 	return RedisSetKeyPrefix
 }
 
-func (r *RedisResourceManager) AllocateResource(ctx context.Context, namespace pluginCore.ResourceNamespace, allocationToken string) (
-	pluginCore.AllocationStatus, error) {
+func (r *RedisResourceManager) checkAgainstOneConstraint(_ context.Context, allAllocated []string,
+	constraint ComposedResourceConstraint) bool {
+	var count int64 = 0
+	for _, allocated := range allAllocated {
+		if does := strings.HasPrefix(allocated, constraint.TargetedPrefixString); does {
+			count++
+		}
+		if count >= constraint.Value {
```

Comment:
Nit: move this line/check as a function on the constraint type.

Reply:
Refactored